### PR TITLE
Fix gtests to always pass null-count to set_null_mask calls

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fix calls to the `column::set_null_mask()` function to always pass the null-count value in gtests source files.

Contributes to: https://github.com/rapidsai/cudf/issues/11968

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/reductions/rank_tests.cpp [hunk 1]
```diff
@@ -214,8 +214,8 @@ TYPED_TEST(TypedRankScanTest, StructsWithNullPushdown)
 
   // First, verify that if the structs column has only nulls, all output rows are ranked 1.
   {
-    struct_col->set_null_mask(
-      create_null_mask(12, cudf::mask_state::ALL_NULL));  // Null mask not pushed down to members.
+    // Null mask not pushed down to members.
+    struct_col->set_null_mask(create_null_mask(12, cudf::mask_state::ALL_NULL), struct_col->size());
```
Comment (bdice) [inline cpp/tests/reductions/rank_tests.cpp:218]: It seems fragile to have one length hardcoded as 12 and another written as a column size call. Should we be preferring the size call? Relatedly, can we do anything to not require/not provide this argument when the mask state is all nulls or no nulls?

Comment (davidwendt) [reply cpp/tests/reductions/rank_tests.cpp:218]: I don't see how we can make this easier other than providing some additional member functions. I don't think adding these is worth doing for this especially for the very rare all-nulls case.

File: cpp/tests/transform/nans_to_null_test.cpp [hunk 2]
```diff
@@ -27,11 +27,11 @@ template <typename T>
 struct NaNsToNullTest : public cudf::test::BaseFixture {
   void run_test(cudf::column_view const& input, cudf::column_view const& expected)
   {
-    auto got_mask = cudf::nans_to_nulls(input);
+    auto [null_mask, null_count] = cudf::nans_to_nulls(input);
     cudf::column got(input);
-    got.set_null_mask(std::move(*(got_mask.first)));
+    got.set_null_mask(std::move(*(null_mask)), null_count);
```
Comment (ttnghia) [inline cpp/tests/transform/nans_to_null_test.cpp:32]:
```suggestion
    got.set_null_mask(std::move(*null_mask), null_count);
```

**Review by ttnghia — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

**Pull request merged**